From b36a67b962d43b28f1115f96e11c483794a2086a Mon Sep 17 00:00:00 2001
From: piyuesh23 <piyuesh23@qed42.com>
Date: Sun, 20 Aug 2017 14:53:27 +0300
Subject: [PATCH] MMCPA-255: Allow adding duplicates to the queue, since we
 don't know here if the product's final price is updated.

---
 .../modules/acq_promotion/src/AcqPromotionsManager.php             | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
index 927e6a6e1abf..978fa4ea9cdd 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
@@ -373,13 +373,6 @@ public function processPromotions(array $promotions = []) {
         // Get list of skus for which promotions should be detached.
         if (!empty($attached_promotion_skus)) {
           $detach_promotion_skus = array_diff($attached_promotion_skus, $fetched_promotion_skus);
-
-          // Clear away skus that are already attached to current promotion.
-          foreach ($fetched_promotion_sku_attach_data as $key => $promotion_sku_data) {
-            if (in_array($key, $attached_promotion_skus)) {
-              unset($fetched_promotion_sku_attach_data[$key]);
-            }
-          }
         }
 
         // Create a queue for removing promotions from skus.
-- 
2.10.1

